### DISCUSSION TITLE
🍏 Shared MacOS device

### DISCUSSION THREAD

**Discussion by bpmct** [coder/coder#2281] [Announcements]
I set up a Mac Mini as a Coder "dev" server so I can do iOS development from any device (e.g my iPad or Linux Thinkpad). I use local VS Code + VNC for the simulator. Works with Swift and React native.

![Screenshot of mobile dev workflow](https://user-images.githubusercontent.com/22407953/173362269-d0aaffa7-87b7-494b-81c0-cbf73589f060.png)

Here's the template: https://github.com/bpmct/coder-templates/tree/main/shared-mac. Multi-user support is planned. Each workspace will be a new system "user" on the device. When stopped, the proper agent will stop. When deleted, the system user can also be removed.

**Comment by WilliamDuTrendMicro** [comment]
thanks 
I also try the template, but find that the agent does not start well in AWS Mac env. 
is there simialr issue in your side ? 

`
"CODER_AGENT_URL=${data.coder_workspace.me.access_url} CODER_AGENT_AUTH=token CODER_AGENT_TOKEN=${coder_agent.dev.token} nohup /usr/local/bin/coder agent &"
`

**Reply by bpmct** [reply]
Do you have the `coder` CLI installed on the MacOS VM? If not, you'll want to run this on the Mac:

```sh
curl -L https://coder.com/install.sh | sh
```

**Reply by bpmct** [reply]
If you have the CLI installed, are there any logs in `/tmp/coder-agent.log` on the Mac?

**Reply by lapnd** [reply]
I got this error on my side
```sh
 github.com/coder/coder/v2/agent.(*agent).run
                   /home/runner/actions-runner/_work/coder/coder/agent/agent.go:697
             - GET /api/v2/appearance: unexpected status code 401: API key or workspace agent token required, but none provided: Try logging in using 'coder login <url>'
 ```

**Reply by lapnd** [reply]
It works perfectly. It was my typo
Thank you!
